listener/avalon_listener/tcs_workorder_receipt_handler.py

```diff
@@ -82,12 +82,22 @@ def WorkOrderReceiptCreate(self, **params):
             )
         else:
             wo_receipt = self.kv_helper.get("wo-receipts", wo_id)
-            if wo_receipt is None:
+            wo_receipt = json.loads(wo_receipt)
+            if wo_receipt and "workOrderId" not in wo_receipt["params"]:
```

Comment:
Under what scenario will workOrderId be not part of wo_receipt ? My understanding is receipt is for a specific work order. So a receipt does not exists without a work order

Reply:
WorkOrderId is exists if work order receipt is created. Refer the git issues for description  https://github.com/hyperledger/avalon/issues/412